src/App.jsx

```diff
@@ -14,33 +17,80 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [sortfield, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  let visibleGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortfield) {
+      case SORT_BY_ALPHABET:
+        return good1.localeCompare(good2);
+      case SORT_BY_LENGTH:
+        return good1.length - good2.length;
+      default:
+        return '';
```

Comment:
The default case in the sorting function should return 0 instead of an empty string. The sort function expects a number as a return value to determine the order of elements.